server/controllers/finance/journal/index.js

```diff
@@ -452,6 +477,17 @@ function transformColumns(rows, newRecord, oldTransaction) {
     .then(() => rows);
 }
 
+//Check the Valid Update of Date
+function selectTransDate(rows){
```

Comment:
I don't think this function does any validation.  If anything, it should just do this:
```js
const selectTransDate = (rows) => rows[0].trans_date;
```